tools/run_functional_tests.py

```diff
@@ -0,0 +1,238 @@
+#!/usr/bin/env python3
+
+from fileinput import filename
+import pprint
+from colorama import Fore, Style
+import os
+import glob
+import re
+import sys
+import argparse
+
+from time import sleep
+
+import serial
+import serial.tools.list_ports
+
+
+#
+# MARK: - argparse
+#
+
+parser = argparse.ArgumentParser(description='Check functional test status')
+
+parser.add_argument('-p', '--port', metavar='PORT', default='/dev/tty.usbmodem*',
+                    help='serial port path used for the robot')
+
+parser.add_argument('-l', '--loop-delay', metavar='SLEEP_DELAY', default=0.8,
+                    help='time delay')
+
+parser.add_argument('-f', '--fast', metavar='FAST', action=argparse.BooleanOptionalAction,
+                    help='fast check with 0.2s loop delay')
+
+parser.add_argument('-v', '--verbose', metavar='VERBOSE', action=argparse.BooleanOptionalAction,
+                    help='provides additional details about command processing and output')
+
+
+args = parser.parse_args()
+
+#
+# MARK: - Serial
+#
+
+PORTS = glob.glob(args.port)
+SERIAL_PORT = PORTS[0] if (len(PORTS) != 0) else args.port
+
+REBOOT_TIMEOUT = 5.0  # in seconds
+SERIAL_TIMEOUT = 0.1  # in seconds
+
+MAX_GET_LINE_RETRIES = REBOOT_TIMEOUT / SERIAL_TIMEOUT
+
+try:
+    com = serial.Serial(SERIAL_PORT, 115200, timeout=SERIAL_TIMEOUT)
+except serial.serialutil.SerialException as error:
+    print(f"{error}")
+    parser.print_help()
+    sys.exit(1)
+
+print(f"Connected to {com.name}")
+
+#
+# MARK: - Functions
+#
+
+verboseprint = print if args.verbose else lambda *a, **k: None
+
+REPOSITORY_BIN = "_build/LEKA_V1_2_DEV/tests/functional/"
+EXTENSION_BIN = ".bin"
+
+LOOP_SLEEP_DELAY = float(args.loop_delay) if not args.fast else 0.1
+
+FILES_BIN = list()
+TESTS = list()
+
+FAILED_STR = Fore.RED + "❌FAILED" + Style.RESET_ALL
+SUCCESS_STR = Fore.GREEN + "✅SUCCESS" + Style.RESET_ALL
+
+
+class Test:
+    saving_repository = "./fs/usr/test/"
+    saving_extension_txt = ".txt"
+    saving_default_filename = "file_saving"
+    saving_filepath = saving_repository + \
+        saving_default_filename + saving_extension_txt
```

Comment:
`saving` means `épargne`, not `saved file`

what you can use instead is `result`, which gives you:

```suggestion
    result_directory = "./fs/usr/test/"
    result_extension = ".txt"
    result_default_filename = "file_saving"
    result_path = result_directory + \
        result_default_filename + result_extension
```

Reply:
these could also be defined globally if needed